docs/spec/concepts.rst

```diff
@@ -3,6 +3,280 @@
 Type system concepts
 ====================
 
+Static, dynamic, and gradual typing
```

Comment:
First off, this section is excellent, and I learned a few things from reading it.

I have to admit that I got a little lost around "Materialization" on my first read-through because a lot of the terminology around gradual types was new to me. What helped me was to explicitly write out a table like this:

Fully static types | Gradual types
------------------------ | ---------------------
B is a subtype of A | B is assignable to A (or B is a consistent subtype of A)
B is a supertype of A | B is assignable from A (or B is a consistent supertype of A)
B is equivalent to A | B is consistent with A

I wonder if including something like this in the doc would also help other readers.

Reply:
I added this table. I put it after the description of all the relations, since it seemed necessary to present them before presenting the table. Let me know if this location would have been helpful in your initial reading, or if it should come sooner.

Reply:
Yeah, I think that works well!